tests/benchmark/supported_dtypes_benchmark.py

```diff
@@ -0,0 +1,710 @@
+
+    except BaseException as e:
+        LOGGER.debug(f"Error during fitting/sampling with dtype '{dtype}': {e}")
+
+    fit_assertion_message = f"{dtype} is no longer supported by 'SYNTHESIZER_FIT'."
+    if fit_result is False:
+        assert fit_result == previous_fit_result, fit_assertion_message
+
+    sample_assertion_message = f"{dtype} is no longer supported by 'SYNTHESIZER_SAMPLE'."
+    if sample_result is False:
+        assert sample_result == previous_sample_result, sample_assertion_message
+
+
+def convert_values(value, inequality):
+    """Convert the given value based on the specified inequality.
+
+    This function checks the provided value and applies a conversion based on
+    the inequality function. If the value satisfies the inequality condition
+    when compared to 0, it multiplies the value by -1. If the value is `None`,
+    it returns `None`.
+
+    Args:
+        value (numeric):
+            The value to be checked and potentially converted. It can be any numeric type or `None`.
+        inequality (function):
+            A comparison function (e.g., `operator.gt` or `operator.lt`) used to compare the
+            value with 0.
+
+    Returns:
+        numeric or None:
+            The converted value if the inequality holds, or the original value otherwise.
+            Returns `None` if the value is `None`.
+    """
+    if pd.isna(value):
+        return None
+
+    if inequality(value, 0):
+        return value * -1
+
+    return value
+
+
+def _create_single_column_constraint_and_data(constraint, data, dtype, sdtype):
+    constraint_class = constraint.get('constraint_class')
+    _dtype = data.dtypes[dtype]
+    constraint['constraint_parameters']['column_name'] = dtype
+
+    if constraint_class == 'Positive' and sdtype == 'numerical':
+        data[dtype] = data[dtype].apply(convert_values, inequality=np.less)
+    elif constraint_class == 'Negative' and sdtype == 'numerical':
+        data[dtype] = data[dtype].apply(convert_values, inequality=np.greater)
+    elif constraint_class == 'ScalarInequality':
+        lower = 0
+        if sdtype == 'numerical':
+            data[dtype] = data[dtype].apply(convert_values, inequality=np.less)
+
+        elif sdtype == 'datetime':
+            # Make the lowest date to be 1971-01-01
+            lower = '1971-01-01'
+
+        constraint['constraint_parameters']['value'] = lower
+
+    elif constraint_class == 'ScalarRange':
+        if sdtype in ('numerical', 'datetime'):
+            low_value = data[dtype].min()
+            high_value = data[dtype].max()
+            constraint['constraint_parameters']['low_value'] = low_value
+            constraint['constraint_parameters']['high_value'] = high_value
+
+    elif constraint_class == 'FixedIncrements':
+        if sdtype == 'numerical':
+            values = [10, 20, 30, 40]
+            if dtype.startswith('pd'):
+                values.append(None)
```

Comment:
should we also put nans in float columns?

Reply:
Yes makes sense, let me add it for all `floats`.